cdc/processor/sinkmanager/manager_impl.go

```diff
@@ -0,0 +1,334 @@
+					upperBoundTs = resolvedTs
+				}
+
+				return sorter.Position{
+					StartTs:  upperBoundTs - 1,
+					CommitTs: upperBoundTs,
+				}
+			}
+			checkAdvance := slowestTableProgress.nextLowerBoundPos.Compare(getUpperBound())
+			if !(checkAdvance == -1 || checkAdvance == 0) || !m.memQuota.tryAcquire(defaultRequestMemSize) {
+				m.progressHeap.push(slowestTableProgress)
+				// Next time.
+				continue
+			}
+			callback := func(lastWrittenPos sorter.Position) {
+				p := &progress{
+					tableID:           tableID,
+					nextLowerBoundPos: lastWrittenPos.Next(),
+				}
+				m.progressHeap.push(p)
+			}
+
+			t := &tableSinkTask{
+				tableID:              tableID,
+				lowerBound:           slowestTableProgress.nextLowerBoundPos,
+				upperBarrierTsGetter: getUpperBound,
+				tableSink:            tableSink.(*tableSinkWrapper),
+				callback:             callback,
+			}
+			select {
+			case <-m.ctx.Done():
+				return m.ctx.Err()
+			case m.tableTaskChan <- t:
+			}
+		}
+	}
+}
+
+// UpdateReceivedSorterResolvedTs updates the received sorter resolved ts for the table.
+func (m *ManagerImpl) UpdateReceivedSorterResolvedTs(tableID model.TableID, ts model.Ts) {
+	tableSink, ok := m.tableSinks.Load(tableID)
+	if !ok {
+		log.Panic("Table sink not found when updating resolved ts",
+			zap.String("namespace", m.changefeedID.Namespace),
+			zap.String("changefeed", m.changefeedID.ID),
+			zap.Int64("tableID", tableID))
+	}
+	tableSink.(*tableSinkWrapper).updateReceivedSorterResolvedTs(ts)
+}
+
+// UpdateBarrierTs updates the barrier ts of all tables in the sink manager.
+func (m *ManagerImpl) UpdateBarrierTs(ts model.Ts) {
+	// It is safe to do not use compare and swap here.
+	// Only the processor will update the barrier ts.
+	// Other goroutines will only read the barrier ts.
+	// So it is safe to do not use compare and swap here, just Load and Store.
+	if ts <= m.lastBarrierTs.Load() {
+		return
+	}
+	m.lastBarrierTs.Store(ts)
+}
+
+// AddTable adds a table(TableSink) to the sink manager.
+func (m *ManagerImpl) AddTable(tableID model.TableID, startTs model.Ts, targetTs model.Ts) {
+	sinkWrapper := newTableSinkWrapper(
+		m.changefeedID,
+		tableID,
+		m.sinkFactory.CreateTableSink(m.changefeedID, tableID, m.metricsTableSinkTotalRows),
+		tablepb.TableStatePreparing,
+		targetTs,
+	)
+	m.tableSinks.Store(tableID, sinkWrapper)
+	initProgress := &progress{
+		tableID:           tableID,
+		nextLowerBoundPos: sorter.Position{StartTs: startTs - 1, CommitTs: startTs},
+	}
+	m.progressHeap.push(initProgress)
+}
+
+// RemoveTable removes a table(TableSink) from the sink manager.
+func (m *ManagerImpl) RemoveTable(tableID model.TableID) error {
+	tableSink, ok := m.tableSinks.Load(tableID)
+	if !ok {
+		log.Panic("Table sink not found when removing table",
+			zap.String("namespace", m.changefeedID.Namespace),
+			zap.String("changefeed", m.changefeedID.ID),
+			zap.Int64("tableID", tableID))
+	}
+	err := tableSink.(*tableSinkWrapper).close(m.ctx)
```

Comment:
if a table is removed,  what will happen for the sink worker in the backend

Reply:
Good question! For now, nothing happened — no more progress advance.